main.go

```diff
@@ -88,10 +91,29 @@ func main() {
 		klog.Fatal("failed to configure k8s client")
 	}
 
+	// secret manager client, without auth so that authentication can be
+	// re-added on a per-RPC basis for each mount
+	smOpts := []option.ClientOption{
+		option.WithUserAgent(ua),
+		// tell the secretmanager library to not add transport-level ADC since we need to override on a per call basis
+		option.WithoutAuthentication(),
+		// grpc oauth TokenSource credentials require transport security, so this must be set explicitly even though TLS
+		// is used
+		option.WithGRPCDialOption(grpc.WithTransportCredentials(credentials.NewTLS(nil))),
+		option.WithGRPCConnectionPool(5),
```

Comment:
Where is .WithGRPCConnectionPool(5) coming from?

Reply:
Ah, I'm requesting that the provider establish 5 individual connections to secret manager so that when the client is used across many concurrent requests the can be round-robin to different underlying connections. I'll add a comment and make the value a cmdline flag